lib/displayHelpers.js

```diff
@@ -91,4 +91,31 @@ const examplePubkey = (index) => {
   return toBase64(data);
 }
 
-export { abbreviateLongString, printableCoin, printableCoins, exampleAddress, examplePubkey };
+/**
+ * Returns an error message for invalid addresses.
+ * 
+ * Returns null of there is no error.
+ */
+const checkAddress = (input) => {
+  if (!input) return "Empty";
+
+  let data;
+  let prefix;
+  try {
+     ({ data, prefix } = Bech32.decode(input));
+  } catch (error) {
+    return error.toString();
+  }
+
+  if (prefix !== process.env.NEXT_PUBLIC_ADDRESS_PREFIX) {
+    return `Expected address prefix '${process.env.NEXT_PUBLIC_ADDRESS_PREFIX}' but got '${prefix}'`;
+  }
+
+  if (data.length !== 20) {
+    return "Invalid address length in bech32 data. Must be 20 bytes.";
+  }
```

Comment:
This is the length check we had before, but generic for all prefix lengths.

The 45 characters check we had before only works as long as the prefix has the same length (like "cosmos" and "tgrade").

Reply:
🔥